SofaKernel/modules/SofaRigid/JointSpring.h

```diff
@@ -0,0 +1,234 @@
+*                (c) 2006-2018 INRIA, USTL, UJF, CNRS, MGH                    *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#ifndef SOFA_JOINTSPRING_H
+#define SOFA_JOINTSPRING_H
+#include "config.h"
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace interactionforcefield
+{
+
+/// JOINTSPRING
+template<typename DataTypes>
+class JointSpring
+{
+public:
+
+    typedef typename DataTypes::Coord    Coord   ;
+    typedef typename Coord::value_type   Real    ;
+    typedef typename DataTypes::Deriv    Deriv   ;
+    typedef typename DataTypes::VecCoord VecCoord;
+    typedef typename DataTypes::VecDeriv VecDeriv;
+
+    enum { N=DataTypes::spatial_dimensions };
+    typedef defaulttype::Mat<N,N,Real> Mat;
+    typedef defaulttype::Vec<N,Real> Vector;
+
+    /// Constructors
+    JointSpring(int m1 = 0,int m2 = 0,
+                Real softKst = 0, Real hardKst = 10000, Real softKsr = 0, Real hardKsr = 10000, Real blocKsr = 100,
+                Real axmin = -100000, Real axmax = 100000, Real aymin = -100000, Real aymax = 100000, Real azmin = -100000, Real azmax= 100000,
+                Real kd = 0);
+
+    /// Attributes
+    int  m1, m2;                    /// the two extremities of the spring: masses m1 and m2
+    Real kd;                        /// damping factor
+    Vector torsion;                 /// torsion of the springs in axis/angle format
+    Vector lawfulTorsion;           /// projected torsion in allowed angles
+    Vector KT;                      /// linear stiffness
```

Comment:
All these inline comments (above and below) need `///<` to be correctly parsed by Doxygen :wink:

Reply:
I didn't know, thanks.
Fixed now